tensorflow/contrib/go/README.md

```diff
@@ -0,0 +1,162 @@
+
+import (
+	"log"
+
+	"github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+func main() {
+	// These are the input tensors to be used
+	inputSlice1 := [][][]int64{
+		{
+			{1, 2},
+			{3, 4},
+		}, {
+			{5, 6},
+			{7, 8},
+		},
+	}
+	inputSlice2 := [][][]int64{
+		{
+			{9, 10},
+			{11, 12},
+		}, {
+			{13, 14},
+			{15, 16},
+		},
+	}
+
+	// Create the two tensors, the data type is recognized automatically as
+	// also the tensor shape from the input slice
+	t1, err := tensorflow.NewTensor(inputSlice1)
+	if err != nil {
+		log.Fatal("Problem trying create a new tensor, Error:", err)
+	}
+
+	t2, err := tensorflow.NewTensor(inputSlice2)
+	if err != nil {
+		log.Fatal("Problem trying create a new tensor, Error:", err)
+	}
+
+	// Load the graph from the file that we generated from Python on the
+	// previous step
+	graph, err := tensorflow.LoadGraphFromTextFile("/tmp/graph/test_graph.pb")
+	if err != nil {
+		log.Fatal("Problem reading the graph from the text file, Error:", err)
+	}
+
+	// Create the session and extend the Graph
+	s, err := tensorflow.NewSession()
+	if err := s.ExtendGraph(graph); err != nil {
+		log.Fatal("Problem extending the Graph, Error:", err)
+	}
+
+	input := map[string]*tensorflow.Tensor{
+		"input1": t1,
+		"input2": t2,
+	}
+	// Execute the graph with the two input tensors, and specify the names
+	// of the tensors to be returned, for this case just one
+	out, err := s.Run(input, []string{"output"}, nil)
+	if err != nil {
+		log.Fatal("Problem trying to run the saved graph, Error:", err)
+	}
+
+	if len(out) != 1 {
+		log.Fatalf("The expected number of outputs is 1 but: %d returned", len(out))
+	}
+
+	outputTensor := out[0]
+	for x := 0; x < outputTensor.Dim(0); x++ {
+		for y := 0; y < outputTensor.Dim(1); y++ {
+			for z := 0; z < outputTensor.Dim(2); z++ {
+				// Using GetVal we can access to the corresponding positions of
+				// the tensor as if we had been accessing to the positions in a
+				// multidimensional array, for instance GetVal(1, 2, 3) is
+				// equivalent to array[1][2][3] on a three dimensional array
+				val, err := out[0].GetVal(x, y, z)
+				if err != nil {
+					log.Fatal("Error trying to read the output tensor, Error:", err)
+				}
+				if val != inputSlice1[x][y][z]+inputSlice2[x][y][z] {
+					log.Printf(
+						"The sum of the two elements: %d + %d doesn't match with the returned value: %d",
+						inputSlice1[x][y][z], inputSlice2[x][y][z], val)
+				}
+
+				log.Println("The value value on coordinates:", x, y, z, "is:", val)
+			}
+		}
+	}
+}
+```
+
+The previous code creates two Tensors to be processed by the previously
+generated Graph, after the execution returns the Tensor with the result.
+
+### Image Recognition
+
+This is a complete code example that shows how to generate Graphs on Go and
+execute them:
+
+[Image Recognition](https://github.com/alonsovidales/tensorflow/blob/go_bindings_tensors/tensorflow/g3doc/tutorials/image_recognition/index.md#usage-with-the-go-api)
```

Comment:
Ups, removed the references now the link paths are relative.
I'm not sure if is correct to leave this examples out of contrib, or if it would be better to relocate them inside contrib/go.